### PR TITLE
Refactors the propagation interfaces, removing the Propagator

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR removes the `Propagation` interface as it is not being used anywhere and it is not described in the specification being added in an attempt of interface segregation.

The move of the constants from the `Tracer` to the `Propagation\Formats` attempts to get more cohesion in the `Propagation` component.

Ping @felixfbecker @beberlei @tedsuo @yurishkuro @pavolloffay

### CODE REVIEW

File: src/OpenTracing/Carriers/TextMap.php [hunk 1]
```diff
@@ -17,7 +17,11 @@ private function __construct(array $textMap)
         }
     }
 
-    public static function create(array $textMap = [])
+    /**
+     * @param array|string[] $textMap
+     * @return TextMap
+     */
+    public static function fromMap(array $textMap = [])
```
Comment (felixfbecker) [inline src/OpenTracing/Carriers/TextMap.php]: I find the term "Map" here confusing, I would think of an SPL `Map`, not an associative array

File: tests/Unit/Carriers/TextMapTest.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ public function testCreateATextMapWithItemsSuccess()
 
     public function testSettingAKeyValuePairSuccess()
     {
-        $textMap = TextMap::create();
+        $textMap = TextMap::fromMap();
```
Comment (felixfbecker) [inline tests/Unit/Carriers/TextMapTest.php]: calling the function `fromMap` without an argument looks weird to me

File: src/OpenTracing/Carriers/TextMap.php (hunk 1 above)
Comment (jcchavezs) [reply src/OpenTracing/Carriers/TextMap.php]: Makes totally sense. I just changed it.

File: tests/Unit/Carriers/TextMapTest.php (hunk 2 above)
Comment (jcchavezs) [reply tests/Unit/Carriers/TextMapTest.php]: I just changed it, this case is only for the sake of testing `set`, ideally you create a `textMap` with values already.

File: src/OpenTracing/Carriers/TextMap.php [hunk 3]
```diff
@@ -17,7 +17,11 @@ private function __construct(array $textMap)
         }
     }
 
-    public static function create(array $textMap = [])
+    /**
+     * @param array|string[] $textMap
+     * @return TextMap
+     */
+    public static function fromArray(array $textMap)
```
Comment (jcchavezs) [inline src/OpenTracing/Carriers/TextMap.php]: @felixfbecker

Comment (felixfbecker) [reply src/OpenTracing/Carriers/TextMap.php]: Probably not, we can add it later if people want it

**Pull request merged**